zeroex/ordervalidator/order_validator.go

```diff
@@ -782,6 +789,28 @@ func (o *OrderValidator) isSupportedAssetData(assetData []byte) bool {
 	return true
 }
 
+func (o *OrderValidator) isSupportedStaticCallData(staticCallAssetData zeroex.StaticCallAssetData) bool {
+	staticCallDataName, err := o.assetDataDecoder.GetName(staticCallAssetData.StaticCallData)
+	if err != nil {
+		return false
+	}
+	switch staticCallDataName {
+	case "checkGasPrice":
+		var decodedStaticCallData zeroex.CheckGasPriceStaticCallData
+		err := o.assetDataDecoder.Decode(staticCallAssetData.StaticCallData, &decodedStaticCallData)
+		if err != nil {
+			return false
+		}
+		// We currently restrict the `checkGasPrice` staticcall to the known MaximumGasPrice contract.
+		if o.contractAddresses.MaximumGasPrice == constants.NullAddress || staticCallAssetData.StaticCallTargetAddress != o.contractAddresses.MaximumGasPrice {
+			return false
+		}
```

Comment:
We only check the contract address here. Where do we validate that it's calling one of the two functions expected? If we aren't validating the function and only the contract, then let's remove the function ABI's from the ABI decoder.

Reply:
We validate that by decoding the `staticCallData`. We first get the name of the staticcall that is encoded by the `staticCallData`. Then if it is `checkGasPrice` (the name of one of our two functions), we actually decode the data to ensure that it is a valid `checkGasPrice` staticcall.

It's only when we've confirmed that the `staticCallData` encodes a valid `checkGasPrice` staticcall that we check the contract address. Is there something that I'm missing about the ABI Decoder? My assumption was that I could use its `Decode` function to validate that a given `staticCallData` satisfies a given ABI.

Reply:
Ah yep, that sounds good enough, I must have just misread. Thanks for clarifying.